File: app/src/main/java/me/echeung/moemoekyun/client/model/Artist.java
package me.echeung.moemoekyun.client.model;

import java.util.List;

import lombok.Getter;

@Getter
public class Artist {
    private List<Album> albums;
    private int id;
    private String image;
    private String lastPlayed;
    private String name;
    private String nameRomaji;
    private long played;
    private String slug;
    private List<Object> songs;

    @Getter
    public static class Album {
        private ArtistAlbum artistAlbum;
        private int id;
        private String image;
        private String name;
        private String nameRomaji;
        private String releaseDate;
        private int type;

        @Getter
        public class ArtistAlbum {
            private int artistId;
            private int albumId;
        }
    }

    @Getter
    public static class Song {
        private List<AlbumSummary> albums;
        private ArtistSong artistSong;
        private List<ArtistSummary> artists;
        private int duration;
        private int id;
        private String lastPlayed;
        private String snippet;
        private String title;
        private String titleRomaji;
        private User uploader;

        @Getter
        public static class AlbumSummary {
            private int albumId;
            private int songId;
            private int trackNumber;
        }

        @Getter
        public static class ArtistSong {
            private int artistId;
            private int songId;
        }

        @Getter
        public static class ArtistSummary {
            private ArtistSong artistSong;
            private int id;
            private String name;
            private String nameRomaji;
        }
    }
}

File: app/src/main/java/me/echeung/moemoekyun/client/model/Artist.kt
package me.echeung.moemoekyun.client.model

class Artist {
    var albums: List<Album>? = null
    var id: Int = 0
    var image: String? = null
    var lastPlayed: String? = null
    var name: String? = null
    var nameRomaji: String? = null
    var played: Long = 0
    var slug: String? = null
    var songs: List<Any>? = null

    class Album {
        var artistAlbum: ArtistAlbum? = null
        var id: Int = 0
        var image: String? = null
        var name: String? = null
        var nameRomaji: String? = null
        var releaseDate: String? = null
        var type: Int = 0

        inner class ArtistAlbum {
            var artistId: Int = 0
            var albumId: Int = 0
        }
    }

    class Song {
        var albums: List<AlbumSummary>? = null
        var artistSong: ArtistSong? = null
        var artists: List<ArtistSummary>? = null
        var duration: Int = 0
        var id: Int = 0
        var lastPlayed: String? = null
        var snippet: String? = null
        var title: String? = null
        var titleRomaji: String? = null
        var uploader: User? = null

        class AlbumSummary {
            var albumId: Int = 0
            var songId: Int = 0
            var trackNumber: Int = 0
        }

        class ArtistSong {
            var artistId: Int = 0
            var songId: Int = 0
        }

        class ArtistSummary {
            var artistSong: ArtistSong? = null
            var id: Int = 0
            var name: String? = null
            var nameRomaji: String? = null
        }
    }
}
